### DISCUSSION TITLE
OSR模式下，网页悬浮效果无效

### DISCUSSION THREAD

**Discussion by njuzhou** [CefView/QCefView#166] [Q&A]
在使用CEF嵌入后，发现网页端的悬浮效果，如果使用title会无效，使用其他方式，如tooltip和css样式，则能够正常显示。(NCW模式下无此问题)
网上搜索了一些办法，有setAttribute(Qt::WA_Mapped)，有添加manifest文件的，都没太有效果

**Comment by tishion** [comment]
@njuzhou 悬浮效果是指的什么？

**Reply by njuzhou** [reply]
悬浮是网页自身的鼠标悬浮效果tooltip，
环境:             QT5.15.2+VS2019+OSR
网页实现:     <div title="xxxx">测试title</div>
已尝试方案: 网上的主要方案是在源代码处增加manifest，指定comctl32.dll为6.0版本，但尝试后并没有效果。此问题在Ncw模式下不存在.

**Reply by njuzhou** [reply]
补充网页实现(源代码没显示出来)
![IMG_20221026_093126](https://user-images.githubusercontent.com/25097261/197913636-90eaf2f6-7fcf-4647-a143-0213951d6c64.jpg)

**Reply by tishion** [reply]
你说的是tooltip，tooltip的跟这个回调有关
https://github.com/CefView/QCefView/blob/8374b33669410659ac4a27fce0aea53f824447ec/src/details/CCefClientDelegate_DisplayHandler.cpp#L156

在NCW模式下，QCefView会自动放过，交给CEF自己处理。
在OSR模式下，这里直接放过，但是CEF不会处理，如果想通过原生方式实现，就要在这里实现，不过QCefView这样处理的原因就是在OSR模式下不支持原生tooltip，而是希望用户通过前端来实现，参考这里
https://www.w3schools.com/css/tryit.asp?filename=trycss_tooltip

**Reply by njuzhou** [reply]
感谢作者大佬！根据上面的信息找了一下，在bitbucket上发现很多人提了类似问题，貌似osr上想显示原生tooltip还很复杂，很多类似问题都没有被解决。(因为嵌入的是存量系统，想让他们因为此问题更改悬浮的实现可能比较困难)
